### DISCUSSION TITLE
I3-gaps dmenu

### DISCUSSION THREAD

**Discussion by plitter** [i3/i3#4701] [Q&A] [answered]
I have a laptop and a hub with some screens with varying sizes connected. When I reconnect the laptop to the hub and try to use dmenu on the main screen it fails, but it works on one of the side screens. This is the error message that I get:
```
$ echo -e 'foo\nbar' | dmenu -p hello
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  1 (X_CreateWindow)
  Value in failed request:  0x0
  Serial number of failed request:  47
  Current serial number in output stream:  50
```
Anyone have any good ideas?

**Comment by Airblader** [comment, marked as answer]
This is an X error coming from dmenu, i.e. i3 is not involved here. This should be either a bug in dmenu or an issue with the X server itself.

You could try running the command under `xtrace` (sometimes shipped as `x11trace`) to see the failing request in detail.

**Reply by plitter** [reply]
I tried with xtrace and the only thing that came up was
```
$ xtrace /usr/bin/dmenu --prompt ">>> " <<<"hello\nbar"
Function              File                                                  Line
--------------------------------------------------------------------------------

```
Thanks for the answer. 

I'll see when I get my irc client up and running again to ask in dwm :)

**Reply by Airblader** [reply]
That's the wrong `xtrace` – you'll probably need to look for `x11trace` then.

**Reply by plitter** [reply]
I found the x11trace, it is in the AUR xtrace, but it is moved to x11trace because glibc owns xtrace. 
http://ix.io/3Gsm is what I get from 
```
x11trace dmenu -p "hello" <<<"foo\nbar" 2>&1 | curl -F 'f:1=<-' ix.io
```

**Reply by Airblader** [reply]
The failing request is

```
000:<:0031: 44: Request(1): CreateWindow depth=0x00 window=0x01a00004 parent=0x00000229 x=0 y=0 width=0 height=26 border-width=0 class=CopyFromParent(0x0000) visual=CopyFromParent(0x00000000) value-list={background-pixel=0x00222222 override-redirect=true(0x01) event-mask=KeyPress,Exposure,VisibilityChange}
```

`width=0` means it tries to create a zero-width window, which is not allowed in X.

The trace also indicates that you are using Xinerama. Is that true – and why? :-)

**Reply by plitter** [reply]
It isn't a conscious choice. I might have to look into that then :)

**Reply by plitter** [reply]
I actually don't know how to disable it. According to `xorg.conf`
```
       Option "Xinerama"  "boolean"
              enable or disable XINERAMA extension.  Default is disabled.
```
I checked for Xinerama in `/etc/X11` and found nothing. I tried adding 
```
$ cat xorg.conf.d/90-xinerama-disable.conf
Section "ServerFlags"
        Option "Xinerama" "false"
EndSection
```
restart and `xdypinfo` still shows Xinerama and `/var/log/Xorg.0.log` still has `[   340.914] (II) Initializing extension XINERAMA`. I don't know how to disable it...
